### ISSUE TITLE
Contributortest

### ISSUE THREAD

**Issue by dommynik** [dommynik/srn-semestergruppe#3]
@MathiasKrill 
@David-Koellnberger 
@erdi-007 
@MiJo0623 

Mir ist eben noch die Idee gekommen, wie man das System etwas abwandeln könnte:

Die Schlüsselkästen werden nicht gemeinsam in einer "keychain-file" gespeichert, sondern einzeln. Bedeutet im konkreten Fall:

Die Software liegt unter "C:/SafeStorage/" - in einem Unterordner namens "./keychains/" befinden sich die einzelnen Schlüsselkästen "dommynik.keychain", "mkrill.keychain", usw.

Jeder neue User kann sich einen Keychain generieren lassen, ohne irgendwie in unserem "SafeStorage"-System eingeloggt zu sein. Der neu generierte Keychain wird dann in dem Unterordner als <username>.keychain abgespeichert - wobei es nicht schlimm ist, wenn der keychain einen anderen Namen trägt.

Am Login-Verfahren ändert sich nichts. Eine Änderung gibt es dann noch beim Schlüssel-Teilen: Die geteilten Schlüssel werden nach dem selben Prinzip generiert, werden aber im eigenen Schlüsselkasten behalten. Der zweite User, für den der Schlüssel gedacht ist, kann dann immer noch an seinen Schlüssel - daran ändert sich also auch nichts.

==> Vorteil: Sollte eine keychain-file fehlen, kann niemand auf die Container zugreifen, die mit den keys aus der keychain-file verschlüsselt sind - auch wenn der Schlüssel für ihn freigegeben wurde. Damit können durch das einfache Löschen der keychain alle Fremdschlüssel weggenommen werden. Ein zweiter User hat dann auch keinen Grund, einen fremden keychain zu verändern.

Was sagt ihr dazu?

Achja: Backups sind dann noch einfacher - copy paste!

**Comment by MathiasKrill** [comment]
"Jeder neue User kann sich einen Keychain generieren lassen, ohne irgendwie in unserem "SafeStorage"-System eingeloggt zu sein." 
-Muss es dann einen gast-zugang geben mit dem man neue Keychains anlegt oder ist das Teil der "Registrierung" ?

Gefällt mir so besser als das vorherige in dem der Schlüssel noch an den "anderen" Keychain vergeben wurde.

**Comment by dommynik** [comment]
Nein, es muss keinen Gast-Zugang geben, sondern einen Registrierungsprozess, bei dem nach Usernamen und Passwort gefragt wird. Siehe Konzept.txt
